### ISSUE TITLE
Old measurement reappears after dropping and then restarting

### ISSUE THREAD

**Issue by andybets** [influxdata/influxdb#16947]
Influx Version 1.7.9

I think related to https://github.com/influxdata/influxdb/issues/10263 and https://github.com/influxdata/influxdb/issues/10052, when I issue a:
```drop measurement funding_rate_1h;```
...the measurement dissappears as expected.

But when stopping and restarting influx, it inexplicably returns.

I hope you guys can manage to find this one, it's really annoying!

**Comment by russorat** [comment]
@andybets thanks for opening. looking at the name (`_1h`), can you double check, you don't have any continuous queries, telegrafs, client libs, or kapacitors sending data to that database/rp/measurement?

**Comment by andybets** [comment]
No, I don't have anything like that, despite the name, and this is when just running the influx process. It would have been embarrassing if I had though! :)

**Comment by mariusvollen** [comment]
I'm experiencing a similar issue. I have multiple VM's that have been decommissioned and physically deleted that i cant get rid of in the database. When I drop the series they disappear. Then there should be a background service deletes the data from disk a few hours later. The longest I've waited after dropping the data is a week. The data still re-appears when i restart the service.

```
root@influxdb:~# influx
Connected to http://localhost:8086 version 1.7.9
InfluxDB shell version: 1.7.9
> use telegraf
Using database telegraf
> SELECT mean("usage_idle") FROM "cpu" WHERE ("host" = 'Cloud')
name: cpu
time mean
---- ----
0    99.86233409313806
> drop series where "host" = 'Cloud'
> SELECT mean("usage_idle") FROM "cpu" WHERE ("host" = 'Cloud')
>
```


Restarting influxdb service...

```
root@influxdb:~# influx
Connected to http://localhost:8086 version 1.7.9
InfluxDB shell version: 1.7.9
> use telegraf
Using database telegraf
> SELECT mean("usage_idle") FROM "cpu" WHERE ("host" = 'Cloud')
name: cpu
time mean
---- ----
0    99.86233409313806
> 
```

**Comment by russorat** [comment]
@mariusvollen thank you for the additional information. We will investigate.

**Comment by relishcolouredhat** [comment]
I'm running into this on 1.8.10 

1. write data with telegraf with incorrect fields
2. stop tg, drop db, confirm no data by no return from select(*)....
3. show field keys still contains invalid types :cry:
